graphql/src/test/java/com/linecorp/armeria/server/graphql/GraphqlServiceSubscriptionTest.java

```diff
@@ -78,14 +89,57 @@ void testSubscription() {
 
         assertThat(response.status()).isEqualTo(HttpStatus.NOT_IMPLEMENTED);
         assertThatJson(response.contentUtf8())
-                .withMatcher("errors",
-                             new CustomTypeSafeMatcher<List<Map<String, String>>>("errors") {
-                                 @Override
-                                 protected boolean matchesSafely(List<Map<String, String>> item) {
-                                     final Map<String, String> error = item.get(0);
-                                     final String message = "WebSocket is not implemented";
-                                     return message.equals(error.get("message"));
-                                 }
-                             });
+                .node("errors[0].message")
+                .isEqualTo("Use GraphQL over WebSocket for subscription");
+    }
+
+    @Test
+    void testSubscriptionOverWebSocket() {
+        final WebSocketClient webSocketClient =
+                WebSocketClient.builder(server.uri(SessionProtocol.H1C, SerializationFormat.WS))
```

Comment:
Do you mind testing for both HTTP1/2 for completeness?

In order to enable HTTP2, I think you will need to modify `AbstractGraphqlService#exchangeType` so that `BIDI_STREAMING` is returned when websockets are enabled.